pkg/setting/setting_plugins.go

```diff
@@ -56,37 +76,29 @@ func (cfg *Cfg) readPluginSettings(iniFile *ini.File) error {
 	cfg.ForwardHostEnvVars = util.SplitString(pluginsSection.Key("forward_host_env_vars").MustString(""))
 	disablePreinstall := pluginsSection.Key("preinstall_disabled").MustBool(false)
 	if !disablePreinstall {
-		rawInstallPlugins := util.SplitString(pluginsSection.Key("preinstall").MustString(""))
-		preinstallPlugins := make(map[string]InstallPlugin)
-		// Add the default preinstalled plugins
+		rawInstallPluginsAsync := util.SplitString(pluginsSection.Key("preinstall").MustString(""))
+		preinstallPlugins := cfg.processPreinstallPlugins(rawInstallPluginsAsync)
+		// Add the default preinstalled plugins to pre install plugins async list
 		for _, plugin := range defaultPreinstallPlugins {
 			preinstallPlugins[plugin.ID] = plugin
 		}
 		if cfg.IsFeatureToggleEnabled("grafanaAdvisor") { // Use literal string to avoid circular dependency
 			preinstallPlugins["grafana-advisor-app"] = InstallPlugin{"grafana-advisor-app", "", ""}
 		}
-		// Add the plugins defined in the configuration
-		for _, plugin := range rawInstallPlugins {
-			parts := strings.Split(plugin, "@")
-			id := parts[0]
-			version := ""
-			url := ""
-			if len(parts) > 1 {
-				version = parts[1]
-				if len(parts) > 2 {
-					url = parts[2]
-				}
-			}
 
-			preinstallPlugins[id] = InstallPlugin{id, version, url}
-		}
+		rawInstallPluginsSync := util.SplitString(pluginsSection.Key("preinstall_sync").MustString(""))
+		preinstallPluginsSync := cfg.processPreinstallPlugins(rawInstallPluginsSync)
 		// Remove from the list the plugins that have been disabled
 		for _, disabledPlugin := range cfg.DisablePlugins {
 			delete(preinstallPlugins, disabledPlugin)
+			delete(preinstallPluginsSync, disabledPlugin)
 		}
 		for _, plugin := range preinstallPlugins {
 			cfg.PreinstallPlugins = append(cfg.PreinstallPlugins, plugin)
 		}
+		for _, plugin := range preinstallPluginsSync {
+			cfg.PreinstallPluginsSync = append(cfg.PreinstallPluginsSync, plugin)
+		}
 		cfg.PreinstallPluginsAsync = pluginsSection.Key("preinstall_async").MustBool(true)
```

Comment:
If `preinstall_async` is false, it means that all plugins will need to be installed asynchronously so you need to move the list of `cfg.PreinstallPlugins` to `cfg.PreinstallPluginsSync`. Please also add a test for that case 🙏